packages/aws-cdk-lib/aws-cloudfront/lib/distribution.ts

```diff
@@ -606,12 +606,20 @@ export class Distribution extends Resource implements IDistribution {
   /**
    * Attach WAF WebACL to this CloudFront distribution
    *
+   * WebACL must be in the us-east-1 region
+   *
    * @param webAclId The WAF WebACL to associate with this distribution
    */
   public attachWebAclId(webAclId: string) {
     if (this.webAclId) {
       throw new Error('A WebACL has already been attached to this distribution');
     }
+    if (webAclId.startsWith('arn:')) {
```

Comment:
Isn't this validation also necessary in the place where the webacl props is being passed?

https://github.com/aws/aws-cdk/blob/28698b82d1a451d90d6ff9303d86c2194fa374ab/packages/aws-cdk-lib/aws-cloudfront/lib/distribution.ts#L342